contrib/kubespray/doc/step-by-step-k8s.md

```diff
@@ -0,0 +1,450 @@
+    node2:
+      ip: x.x.x.38
+      access_ip: x.x.x.38
+      ansible_host: x.x.x.38
+      ansible_ssh_pass: "your-password-here"
+      ansible_ssh_extra_args: '-o StrictHostKeyChecking=no'
+    node3:
+      ip: x.x.x.39
+      access_ip: x.x.x.39
+      ansible_host: x.x.x.39
+      ansible_ssh_pass: "your-password-here"
+      ansible_ssh_extra_args: '-o StrictHostKeyChecking=no'
+```
+
+- Notice: 
+   - OpenPAI need 3 master nodes. An etcd cluster will be setup on the three nodes, and one of them will be master node of kubernetes. 
+   - We assume all master nodes in your cluster is none-gpu machine.
+
+###### ```worker.yaml for worker (GPU) node```
+
+```yaml
+all:
+  hosts:
+    node4:
+      ip: x.x.x.40
+      access_ip: x.x.x.40
+      ansible_host: x.x.x.40
+      ansible_ssh_pass: "your-password-here"
+      ansible_ssh_extra_args: '-o StrictHostKeyChecking=no'
+    node5:
+      ip: x.x.x.41
+      access_ip: x.x.x.41
+      ansible_host: x.x.x.41
+      ansible_ssh_pass: "your-password-here"
+      ansible_ssh_extra_args: '-o StrictHostKeyChecking=no'
+    node6:
+      ip: x.x.x.42
+      access_ip: x.x.x.42
+      ansible_host: x.x.x.42
+      ansible_ssh_pass: "your-password-here"
+      ansible_ssh_extra_args: '-o StrictHostKeyChecking=no'
+```
+
+###### Prepare ansible environment
+
+```bash
+sudo apt update
+sudo apt-get -y install software-properties-common python3 python3-dev
+
+curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+sudo python3 get-pip.py
+
+sudo pip3 install paramiko
+
+sudo apt-get install sshpass
+
+git clone https://github.com/kubernetes-sigs/kubespray.git
+cd kubespray
+git checkout release-2.11
+sudo pip3 install -r requirements.txt
+```
+
+###### The cluster has been deployed OpenPAI before
+
+- If you deployed OpenPAI before, especially the yarn version OpenPAI bfore. Please refer to the following link to clean up your cluster before the next steps.
+    - [Guid to Clean Your Environment](./clean-env.md)
+- If you haven't deployed openpai in your cluster before, you could step to next.
+
+
+###### Ansible test
+
+```bash
+
+ansible all -i host.yml -m ping
+
+```
+
+###### Create docker configuration for OpenPAI
+
+```shell script
+ansible-playbook -i /path/to/infra.yml copy-daemon-openpai-default-runtime.yml --become --become-user=root 
+
+ansible-playbook -i /path/to/worker.yml copy-daemon-openpai-nvidia-runtime.yml --become --become-user=root 
+```
+
+- Notice
+
+    We assume that your infra nodes don't have GPU card. So in our tutorial, nvidia drivers and nvidia runtime isn't installed on the infra node.
+    If your infra nodes are GPU machines, and you wanna install driver and runtime on those machine. You should do the same steps for your infra nodes as what you have done for the worker nodes.
+    And be careful, at the last step named ```create docker configuration for OpenPAI```, you should use the playbook named ```copy-daemon-openpai-nvidia-runtime.yml``` for your infra node.
```

Comment:
suggest update to:
One more important note (if you are using GPU nodes as infra nodes): **make sure you are using** ```copy-daemon-openpai-nvidia-runtime.yml``` at the last step instead of the default file for non-GPU nodes ```create docker configuration for OpenPAI```.